Assets/UI/toppanel_CQUI.lua

```diff
@@ -9,78 +9,71 @@ BASE_CQUI_RefreshResources = RefreshResources;
+    local infoStackX:number = Controls.StaticInfoStack:GetSizeX();
+    local metaStackX:number = Controls.RightContents:GetSizeX();
     local screenX, _:number = UIManager:GetScreenSizeVal();
-    local maxSize = screenX - yieldStackX - infoStackX - metaStackX - META_PADDING;
-    if (maxSize < 0) then maxSize = 0; end
-    local currSize = 0;
-    local isOverflow = false;
-    local overflowString = "";
-    local plusInstance:table;
+    local maxSize:number = math.max(screenX - yieldStackX - infoStackX - metaStackX - META_PADDING, 0);
+    local currSize:number = 0;
+    local isOverflow:boolean = false;
+    local overflowString:string = "";
+    local plusInstance:table = nil;
 
     -- CQUI/jhcd: show RESOURCECLASS_LUXURY too, if it is enabled in CQUI settings
-    if (CQUI_showLuxury) then
-      for resource in GameInfo.Resources() do
-        if (resource.ResourceClassType ~= nil and resource.ResourceClassType ~= "RESOURCECLASS_BONUS" and resource.ResourceClassType ~= "RESOURCECLASS_STRATEGIC" and resource.ResourceClassType ~="RESOURCECLASS_ARTIFACT") then
-          local amount = pPlayerResources:GetResourceAmount(resource.ResourceType);
-          if (amount > 0) then
-            local resourceText = "[ICON_"..resource.ResourceType.."] ".. amount;
-            local numDigits = 3;
-            if (amount >= 10) then
-              numDigits = 4;
-            end
-            local guessinstanceWidth = math.ceil(numDigits * FONT_MULTIPLIER);
-            if(currSize + guessinstanceWidth < maxSize and not isOverflow) then
-              if (amount ~= 0) then
-                local instance:table = m_kResourceIM:GetInstance();
-                instance.ResourceText:SetText(resourceText);
-                instance.ResourceText:SetToolTipString(Locale.Lookup(resource.Name).."[NEWLINE]"..Locale.Lookup("LOC_TOOLTIP_LUXURY_RESOURCE"));
-                instanceWidth = instance.ResourceText:GetSizeX();
-                currSize = currSize + instanceWidth;
-              end
-            else
-              if (not isOverflow) then
-                overflowString = amount.. "[ICON_"..resource.ResourceType.."]".. Locale.Lookup(resource.Name);
-                local instance:table = m_kResourceIM:GetInstance();
-                instance.ResourceText:SetText("[ICON_Plus]");
-                plusInstance = instance.ResourceText;
-              else
-                overflowString = overflowString .. "[NEWLINE]".. amount.. "[ICON_"..resource.ResourceType.."]".. Locale.Lookup(resource.Name);
-              end
-              isOverflow = true;
-            end
-          end
-        end
-      end
-    end
+    if CQUI_showLuxury then
+        for resource in GameInfo.Resources() do
+            if resource.ResourceClassType == "RESOURCECLASS_LUXURY" then -- it cleary says "show luxuries in the top panel"...
+                local amount:number = pPlayerResources:GetResourceAmount(resource.ResourceType);
```

Comment:
Unmodded game has a null check for `resource.ResourceClassType`. I'm guessing your removal for that means it was unnecessary, is that right?

Reply:
This field has a NOT NULL constraint in DB. It cannot be nil.